fcrepo/fcrepo
# Switch RDF processing to use external URIs.

**JIRA Ticket**: https://jira.lyrasis.org/browse/FCREPO-3598

# What does this Pull Request do?
When a relative URI (ie. `</rest/someObject>`) is used in a RDF document it gets parsed against the baseUri and ends up as `info:/rest/someObject`. This is fine if using the internal jetty as there is no additional context path, but when deployed to Tomcat under (say `fcrepo6`) you get incorrect internal IDs.

This now takes the context path into consideration.

# How should this be tested?

1. Before the PR, deploy F6 to Tomcat under whatever context (ie `fcrepo6` for example).
1. Try to PUT a body using a valid relative URI.
     ```
     curl -i -ufedoraAdmin:fedoraAdmin -XPUT -H"Content-type: text/turtle" -d "<> <http://example.org/related> </fcrepo6/rest/object>" http://localhost:8080/fcrepo6/rest/testObject
     ```
1. Get the new object and see that the `/fcrepo6/rest/object` is not correctly converted and appears as `info:/fcrepo6/rest/object`
1. Pull in and build this PR.
1. Try the above PUT statement again with a new resource.
      ```
     curl -i -ufedoraAdmin:fedoraAdmin -XPUT -H"Content-type: text/turtle" -d "<> <http://example.org/related> </fcrepo6/rest/object>" http://localhost:8080/fcrepo6/rest/anotherObject
     ```
1. Get this object and see that `/fcrepo6/rest/object` now is converted to `http://localhost:8080/fcrepo6/rest/object`.
1. This should also work under any context and under the jetty container with just `/rest`


# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo/committers
